.github/workflows/continuous-integration.yml

```
@@ -33,37 +30,11 @@ jobs:
 
     steps:
 
-    - name: Free Disk Space (Ubuntu)
-      uses: jlumbroso/free-disk-space@v1.3.1
-      if: runner.os == 'Linux'
-      with:
-        # this might remove tools that are actually needed,
-        # if set to "true" but frees about 6 GB. Currently we don't seem to need
-        # this extra disk space
-        tool-cache: false
-
-        # we might turn some of those off to trade CI time for less space saving
-        # if this step turns out to be too slow
-        android: true
-        dotnet: true
-        haskell: true
-        large-packages: true
-        docker-images: true
-        swap-storage: true
-
     - name: Checking out the repository
       uses: actions/checkout@v4
       with:
         fetch-depth: 0
 
-    - name: Installing Nix
-      uses: cachix/install-nix-action@V27
-      with:
-        nix_path: nixpkgs=channel:nixos-unstable
-        extra_nix_config: |
-          experimental-features = nix-command flakes
-          accept-flake-config = true
-
     - name: Setup Cachix
       uses: cachix/cachix-action@v15
       with:
```

`cachix/cachix-action` [uses](https://github.com/cachix/cachix-action/blob/master/src/main.ts#L26) `nix-env` to install `cachix` binary. It seems like runner user's profile is not included in PATH by default, so `which cachix` [check](https://github.com/cachix/cachix-action/blob/master/src/main.ts#L62) after that fails. You could fetch and install this binary using this flake's `nixpkgs` input like this:

```suggestion
      - name: Install cachix
        id: install_cachix
        run: |
          set -euo pipefail
          cachix_path="$(nix build --print-out-paths --inputs-from . nixpkgs#cachix.out)"
          echo "cachixBin=${cachix_path}/bin/cachix" >> "$GITHUB_OUTPUT"

        - name: Setup Cachix
      uses: cachix/cachix-action@v15
      with:
        cachixBin: ${{ steps.install_cachix.outputs.cachixBin }}
```

(I'd put the last line after other inputs, but GitHub won't let me)

You also get pinned `cachix` binary with this :)